openapi/nexus-internal.json

```diff
@@ -3128,6 +3128,7 @@
         "type": "object",
         "properties": {
           "address": {
+            "nullable": true,
```

Comment:
Before this PR: The public-facing API to "add a dataset to the DB" required a SocketAddr type.

This is an artifact which was partially fixed by https://github.com/oxidecomputer/omicron/issues/2000 - unfortunately, there is code which still **uses** these IP/port pairs if they exist, but they are not required to exist for a dataset record to be created.

Anyway - TL;DR, they're optional, as they should be.